components/camel-kafka/src/main/java/org/apache/camel/processor/idempotent/kafka/KafkaIdempotentRepository.java

```diff
@@ -0,0 +1,344 @@
+import org.apache.camel.util.LRUCache;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.camel.util.StringHelper;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A Kafka topic-based implementation of {@link org.apache.camel.spi.IdempotentRepository}.
+ *
+ * Uses a local cache of previously seen Message IDs. All mutations of the cache are via a Kafka topic, on which
+ * additions and removals are broadcast. The topic used must be unique per repository instance. This class makes no
+ * assumptions about number of partitions (it is designed to consume from all at the same time), or replication factor.
+ * Each repository instance that uses the topic (e.g. on different machines running in parallel) controls its own
+ * consumer group.
+ *
+ * On startup, the instance subscribes to the topic and rewinds the offset to the beginning, rebuilding the cache to the
+ * latest state.
+ *
+ * To use, this repository must be placed in the Camel registry, either manually or by registration as a bean in
+ * Spring/Blueprint, as it is CamelContext aware.
+ *
+ * @author jkorab
+ */
+@ManagedResource(description = "Kafka IdempotentRepository")
+public class KafkaIdempotentRepository extends ServiceSupport implements IdempotentRepository<String>, CamelContextAware {
+
+    private static final int DEFAULT_MAXIMUM_CACHE_SIZE = 1000;
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    private final Map<String, Object> cache;
+
+    private final AtomicLong duplicateCount = new AtomicLong(0);
+    private final String topic;
+    private final Properties producerConfig;
+    private final Properties consumerConfig;
+
+    private Consumer<String, String> consumer;
+    private Producer<String, String> producer;
+    private TopicPoller topicPoller;
+
+    private CamelContext camelContext;
+    private ExecutorService executorService;
+    private CountDownLatch cacheReadyLatch;
+
+    enum CacheAction {
+        add,
+        remove,
+        clear
+    }
+
+    public KafkaIdempotentRepository(String topic, String bootstrapServers) {
+        this(topic, bootstrapServers, DEFAULT_MAXIMUM_CACHE_SIZE);
+    }
+
+    public KafkaIdempotentRepository(String topic, String bootstrapServers, int maxCacheSize) {
+        StringHelper.notEmpty(topic, "topic");
```

Comment:
Its better to move such logic to doStart and then just let ctr set the values in their setters.

Then you have all init logic once in doStart that check for null / empty parameters and so on